drivers/flash/nrf_qspi_nor.c

```diff
@@ -556,9 +583,29 @@ static int qspi_nor_write(struct device *dev, off_t addr, const void *src,
 		return -EINVAL;
 	}
 
-	qspi_lock(dev);
 
-	nrfx_err_t res = nrfx_qspi_write(src, size, addr);
+	nrfx_err_t res;
+#if IS_ENABLED(NORDIC_QSPI_NOR_FLASH_ALLOW_STACK_USAGE_FOR_DATA_IN_FLASH)
+	if (((u32_t)src < CONFIG_SRAM_BASE_ADDRESS) && !(size < 4U)) {
+		/* if the value is in flash we need to copy it into RAM since
+		 * the DMA won't be able to transport data from flash to QSPI
+		 */
+		src = buf;
+		for (size_t i = 0; i < size ; i += sizeof(buf)) {
```

Comment:
Due to the check above we can be sure that the data is aligned by 4 bytes.

Reply:
Did you consider having a larger buffer? (In case it's significant performance-wise - I don't know :))

Reply:
I was thinking just to reuse the buffer you already have. Since this is an edge case where you could easily work around it by just doing local copy of the data I think it's okei.

If you want speed it's much faster to copy all of your Flash data into a Ram buffer then pass it to the function than relaying on this.

Reply:
Hmm this doesn't seem to work :/

Reply:
Hm, give me a minute I need to checkout your code to see what is happening.

Reply:
Fixed.